docs/react-v9/contributing/testing-with-jest.md

```diff
@@ -0,0 +1,234 @@
+> 🚨🚨 **This page is primarily about `@fluentui/react-components` ("v9") and `@fluentui/react` ("v8") and related packages.** 🚨🚨&nbsp; [See this page for `@fluentui/react-northstar` ("v0").](https://github.com/microsoft/fluentui/blob/master/packages/fluentui/test-a-feature.md)
+
+## Overview
+
+Fluent UI's unit, functional, and snapshot tests are built using [Jest](https://jestjs.io/). This allows us to run tests in a Node environment but simulates the browser using [jsdom](https://github.com/jsdom/jsdom). ([See this page for other types of tests.](Testing))
+
+We use various libraries on top of Jest for rendering and interacting with React components:
+
+- New tests should use [React Testing Library](https://testing-library.com/docs/react-testing-library/intro)
+- Legacy tests may also use:
+  - [Enzyme](https://enzymejs.github.io/enzyme/) -- **not allowed in new tests**
+  - [react-test-renderer](https://reactjs.org/docs/test-renderer.html), primarily for snapshot testing _(snapshots are discouraged in new tests)_
```

Comment:
Let's keep this as "FYI" to not confuse contributors that this is something we use as a standard

```suggestion
> [!NOTE]  
> Old monorepo libraries (like v8)( **non of these approaches is allowed in any new tests** ):
> - [Enzyme](https://enzymejs.github.io/enzyme/) -- **not allowed in new tests**
> - [react-test-renderer](https://reactjs.org/docs/test-renderer.html), primarily for snapshot testing _(snapshots are discouraged in new tests)_
```